### ISSUE TITLE
[Blazor] Newtonsoft.Json.JsonConvert fails to deserialize POCO in WASM environment

### ISSUE THREAD

**Issue by MarkStega** [aspnet/AspNetCore#7958]
### Describe the bug
I have a service that is creating a json representation of a POCO. When I consume the service on the server side the json can be deserialized by NewtonSoft.Json. When consumed in the client the deserialization fails and leaves 'garbage' in the POCO.

### To Reproduce
1. ASP.NET Core '3.0.0 Preview 2'
2. Blazor '0.8.0'
3. Run this code in the wasm client;
                var json2 = "{\"pStatus\":0,\"pError\":\"\",\"pResult\":[]}";
                var srtest = Newtonsoft.Json.JsonConvert.DeserializeObject<ServiceResult<Process_DD[]>>(json2);

4. Examine the srtest content and see that pStatus has a value of '21778960' (It started as an enum of {OK, Error}) and that pResult is null rather than an empty array of Process_DD objects.

### Expected behavior
I expect the deserialization to work on the WASM client as it does on the server.

### Screenshots
n/a

### Additional context
C:\Users\ms>dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview-010184
 Commit:    c57bde4593

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100-preview-010184\

Host (useful for support):
  Version: 3.0.0-preview-27324-5
  Commit:  63a01b08e5

.NET Core SDKs installed:
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.504 [C:\Program Files\dotnet\sdk]
  2.1.600-preview-009472 [C:\Program Files\dotnet\sdk]
  2.1.600-preview-009497 [C:\Program Files\dotnet\sdk]
  3.0.100-preview-010184 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0-preview-19075-0444 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0-preview-27324-5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0-preview-27325-3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download

**Comment by MarkStega** [comment]
Additional information: I've tracked the issue down and have discovered that if I have the data returned from the controller on the server defined as the type Process_DD[] then the deserialization works. If however I return a type of ServiceResult<Process_DD[]> the deserialization fails in the WASM client and works if executed on the server. I've attached the source files for the two POCOs.

[TypeDefinitions.zip](https://github.com/aspnet/AspNetCore/files/2911091/TypeDefinitions.zip)
